samples/hello_world/boards/mimx8mp_evk_itcm.conf

```diff
@@ -0,0 +1 @@
+CONFIG_OPENAMP_RSC_TABLE=y
```

Comment:
Seems that you enable the CONFIG_OPENAMP_RSC_TABLE for bad reason. 
On the stm32mp1 platform I m able to load and run this example without resource table.

No need of the resource table to be able to load the demo, but need should be linked to the redirection of the printk.
On stm32mp1 we use the [RAM CONSOLE](https://github.com/zephyrproject-rtos/zephyr/blob/main/soc/arm/st_stm32/stm32mp1/Kconfig.series#L14) that redirects to the Linux remoteproc trace referenced in the resource table. 

In your case if you output the prink on a serial link, you don't need a resource table

Reply:
After spending some time debugging this issue on the Linux side, I can tell that the crash is caused by the vendor-specific imx_rproc driver.
The implementation in the [vendor kernel](https://source.codeaurora.org/external/imx/linux-imx/tree/drivers/remoteproc/imx_rproc.c?h=lf-5.10.y) has some minor modifications to support additional resource table scenarios not present in the mainline kernel.
These changes are responsible for the crash when loading an ELF without a resource table. With a current mainline kernel (5.17-rc2), the firmware can be loaded without any problems.

Unfortunately, there seem to be some other issues when reloading a firmware with a mainline kernel, causing the system to hang when the remote processor boots.
So, at the moment, only the current vendor kernel seems to work for me, but hopefully that will change soon.

TLDR:
- The mainline kernel should not need an RSC table (OPENAMP_RSC_TABLE is not needed).
- The current vendor kernel from NXP needs an RSC table (this looks like a bug to me, but I can't say if it will be fixed or if this is planned for upstream at all).
- Without OPENAMP_RSC_TABLE the hello world example cannot be used with the reference software (NXP imx yocto project).
- Enabling OPENAMP_RSC_TABLE could be seen as a handy helper for people evaluating Zephyr on i.MX with the reference software

What should I do (besides the requested fixes)?
Should I just delete the 'OPENAMP_RSC_TABLE' commit for the hello world example?

Reply:
Ok, I decided to just delete the commit that enabled the rsc table for the hello world example. If someone wants to load this example with the current nxp vendor kernel, they have to add the table manually.
With a mainline kernel, the rsc table is not needed.